src/ray/object_manager/ownership_based_object_directory.cc

```diff
@@ -253,22 +220,33 @@ ray::Status OwnershipBasedObjectDirectory::SubscribeObjectLocations(
     const rpc::Address &owner_address, const OnLocationsFound &callback) {
   auto it = listeners_.find(object_id);
   if (it == listeners_.end()) {
-    WorkerID worker_id = WorkerID::FromBinary(owner_address.worker_id());
-    std::shared_ptr<rpc::CoreWorkerClient> rpc_client = GetClient(owner_address);
-    if (rpc_client == nullptr) {
-      RAY_LOG(WARNING) << "Object " << object_id << " does not have owner. "
-                       << "SubscribeObjectLocations becomes a no-op.";
-      return Status::OK();
-    }
-    rpc::GetObjectLocationsOwnerRequest request;
-    request.set_intended_worker_id(owner_address.worker_id());
-    request.set_object_id(object_id.Binary());
-    request.set_last_version(-1);
-    rpc_client->GetObjectLocationsOwner(
-        request,
-        std::bind(&OwnershipBasedObjectDirectory::SubscriptionCallback, this, object_id,
-                  worker_id, std::placeholders::_1, std::placeholders::_2));
-    it = listeners_.emplace(object_id, LocationListenerState()).first;
+    // Create a object eviction subscription message.
```

Comment:
Supernit: I can't help myself.
```suggestion
    // Create an object eviction subscription message.
```